test/core/test/fixture-concurrent-beforeEach.test.ts

```diff
@@ -0,0 +1,45 @@
+import { afterAll, beforeEach, expect, test } from 'vitest'
+
+// this test case might look exotic, but a few conditions were required to reproduce the reported bug.
+// such particular conditions are marked with "[repro]" in the comments.
+
+let globalA = 0
+let globalB = 0
+
+interface MyFixtures {
+  a: number
+  b: number
+}
+
+export const myTest = test.extend<MyFixtures>({
+  // [repro] fixture order must be { a, b } and not { b, a }
+  a: async ({}, use) => {
+    globalA++
+    await new Promise<void>(resolve => setTimeout(resolve, 200)) // [repro] async fixture
+    await use(globalA)
+  },
+  b: async ({}, use) => {
+    globalB++
+    await use(globalB)
+  },
+})
+
+// [repro] beforeEach uses only "b"
+beforeEach<MyFixtures>(({ b }) => {
+  expect(b).toBeTypeOf('number')
+})
+
+afterAll(() => {
+  expect([globalA, globalB]).toEqual([2, 2])
+})
+
+// [repro] concurrent test uses both "a" and "b"
+myTest.concurrent('test1', async ({ a, b }) => {
+  expect(a).toBeTypeOf('number')
+  expect(b).toBeTypeOf('number')
+})
+
+myTest.concurrent('test2', async ({ a, b }) => {
+  expect(a).toBeTypeOf('number')
+  expect(b).toBeTypeOf('number')
+})
```

Comment:
What's happening was that, due to concurrency:

- beforeEach: setup "b" fixture for "test1" context
- beforeEach: setup "b" fixture for "test2" context
- "test1" setup "a" fixture
- `callFixtureCleanup` for "test1" (but actually it also cleans up "b" fixture for "test2")
- "test2" setup "a" fixture and "b" fixture